app/src/main/java/org/liberty/android/fantastischmemo/entity/Card.java

```diff
@@ -191,4 +195,36 @@ public boolean equals(Object c) {
     public int hashCode() {
         return getId();
     }
+
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel parcel, int i) {
+        parcel.writeString(this.getQuestion());
+        parcel.writeString(this.getAnswer());
+        parcel.writeString(this.getHint());
+        parcel.writeInt(this.getId());
+    }
+
+    public static final Parcelable.Creator<Card> CREATOR
```

Comment:
What exactly is the parcel?

Reply:
To store an ArrayList<Card> in the intent, we need to store it in a parcel. To do this, we need to make the Card class parcelable. The code above is how we make the Card class parcelable.

If you look in the GradeButtonsFragment class, we obtain a parcel, write a value to it, and then we set the parcel in the Card class for the current card.

Later, when we are getting the intent in the QuizReviewActivity we are retrieving all the information that was stored in the parcels.